index.js

```diff
@@ -480,7 +481,16 @@ class MagicOccupancy {
     }
 
     _setOccupancyState (newVal) {
+        // Capture the previous state and determine if the state change should be logged
+        this.previousModeState = this.modeState;
+        this.occupancyLoggingCondition = (newVal == 'Occupied' && this.previousModeState == 'Unoccupied' || newVal == 'Unoccupied')
```

Comment:
Last little thing - this shouldn't be a class-wide variable. I would probably use const here instead

```
const occupancyLoggingCondition = (newVal == 'Occupied' && this.previousModeState == 'Unoccupied' || newVal == 'Unoccupied')
if (this.occupancyLogging && occupancyLoggingCondition) {

```

Reply:
That makes sense. Thanks for the suggestion! The change has been made in dc54e03.